lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -205,10 +208,17 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             merge=True,
         )
 
-        # currently only make sure there is a CA
-        # TODO: workflow for replacement will be added later
-        if self._read_stored_ca() is None:
-            self.store_new_ca(self.charm.secrets.get_object(scope, cert_type.val))
+        current_stored_ca = self._read_stored_ca()
+        if current_stored_ca != event.ca:
+            if not self.store_new_ca(self.charm.secrets.get_object(scope, cert_type.val)):
+                logger.debug("Could not store new CA certificate.")
+                return
```

Comment:
Should we defer here? or how do we react if the storing of the new CA didn't succeed?

Reply:
Yes, I will add that.

Reply:
Done.